include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -1446,23 +1757,27 @@ __parallel_set_scan(oneapi::dpl::__internal::__device_backend_tag __backend_tag,
 }
 
 template <typename _CustomName, typename _Range1, typename _Range2, typename _Range3, typename _Compare,
-          typename _IsOpDifference>
+          typename _SetTag>
 __future<sycl::event, __result_and_scratch_storage<oneapi::dpl::__internal::__difference_t<_Range3>>>
 __parallel_set_op(oneapi::dpl::__internal::__device_backend_tag __backend_tag, sycl::queue& __q, _Range1&& __rng1,
-                  _Range2&& __rng2, _Range3&& __result, _Compare __comp, _IsOpDifference __is_op_difference)
+                  _Range2&& __rng2, _Range3&& __result, _Compare __comp, _SetTag __set_tag)
 {
-    if (oneapi::dpl::__par_backend_hetero::__is_gpu_with_reduce_then_scan_sg_sz(__q))
+    if constexpr (_SetTag::__can_write_from_rng2_v)
     {
-        return __parallel_set_reduce_then_scan<_CustomName>(
-            __backend_tag, __q, std::forward<_Range1>(__rng1), std::forward<_Range2>(__rng2),
-            std::forward<_Range3>(__result), __comp, __is_op_difference);
-    }
-    else
-    {
-        return __parallel_set_scan<_CustomName>(__backend_tag, __q, std::forward<_Range1>(__rng1),
-                                                std::forward<_Range2>(__rng2), std::forward<_Range3>(__result), __comp,
-                                                __is_op_difference);
+        return __parallel_set_reduce_then_scan<_CustomName>(__backend_tag, __q, std::forward<_Range1>(__rng1),
+                                                            std::forward<_Range2>(__rng2),
+                                                            std::forward<_Range3>(__result), __comp, __set_tag);
     }
+    return __parallel_set_scan<_CustomName>(__backend_tag, __q, std::forward<_Range1>(__rng1),
+                                            std::forward<_Range2>(__rng2), std::forward<_Range3>(__result), __comp,
+                                            __set_tag);
+}
+
+template <typename _ExecutionPolicy>
+bool
+__can_set_op_write_from_set_b(_ExecutionPolicy&& __exec)
```

Comment:
Probably we should add first param into this function - `oneapi::dpl::__internal::__device_backend_tag`
And on caller side - pass `_BackendTag{}`

Reply:
thanks, done.